lib/appium.js

```diff
@@ -176,15 +178,26 @@ class AppiumDriver extends BaseDriver {
       }
     }
 
-    let curSessions;
+    let runningDriversData;
     try {
-      curSessions = this.curSessionDataForDriver(InnerDriver);
+      runningDriversData = this.curSessionDataForDriver(InnerDriver);
     } catch (e) {
       throw new errors.SessionNotCreatedError(e.message);
     }
 
+    let innerSessionId, dCaps;
     let d = new InnerDriver(this.args);
-    let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
+    try {
+      if (_.isArray(this.pendingDrivers[InnerDriver.name])) {
```

Comment:
Will `this.pendingDrivers[InnerDriver.name]` ever be anything other than `undefined` or an `Array`? This could just be
```js
this.pendingDrivers[InnerDriver.name] = this.pendingDrivers[InnerDriver.name] || [];
this.pendingDrivers[InnerDriver.name].push(d);
```